### ISSUE TITLE
Payments from friends/family parties not handled correctly in SF

### ISSUE THREAD

**Issue by mscholtz** [washingtontrails/vms#254]
When a payment from a f/f party is made via VMS, and sent to C&P as a single transaction w/ multiple line items, we have the following problems:

1. C&P creates two separate opps (since it always does that w/ line items unless we have Opp Products turned on
2. Both opps are getting tagged w/ the Registration ID of the purchaser
3. This leads to only the purchaser's reg getting the Registered status.  The registrations of the other party members are staying in "Payment Not Provided"

We considered turning on the Product-based processing in C&P, but Charlie decided against that for now.  Instead, we will implement the following quicker workaround:

* [x] change what VMS passes to C&P to combine all the registrations into 1 line item.  this should lead to 1 opp being created in SF, w/ the registration ID of the payor
* [ ] change the SF opp trigger to check if the registration has an associated f/f party.  if so, assume that the payment covered the entire party, and make the status of all party members Registered

Charlie understands that this solution will mean less detail available for reporting in SF.

**Comment by mscholtz** [comment]
@WaBirder57 here's a ticket for what we discussed yesterday
@davisagli I think we're ready for your part of this
@ninetaLogik can you do the SF side once David's done?  Let me know if you want to discuss it.  Let's do this before story 194

**Comment by mscholtz** [comment]
@davisagli while you're at it, why don't you pass a custom field w/ the donation amount if there is a separate donation, for story 194.  It's now clear that's what we need for that.  Call it whatever you want, just let us know the name.

**Comment by davisagli** [comment]
Okay I'll make sure Carlos knows to do that as part of story 194

**Comment by davisagli** [comment]
@mscholtz @ninetaLogik ok I updated the pyramid app to always pass a single line item with the total amount.

**Comment by ninetaLogik** [comment]
@mscholtz I tested and verified that if we have multiple Payment Pending Regs (such as one person registering multiple people) the code works correctly and when the Fee/Purchase opp comes in all related regs get Status=Registered. We designed this trigger to work one-at-a-time, and my suspicion is that since we're now getting multiple c&p opps our trigger does not handle that. I will verify my assumption.
